src/components/AssetsComponent.jsx

```diff
@@ -20,46 +21,66 @@ export default class AssetsComponent extends Component {
   };
 
   componentDidMount() {
-    const { activePage, assetsList } = this.props;
-
-    const assetsEmpty = isEmpty(assetsList);
-
     // TODO: fix the logic so that assets are fetched when you create an asset before fetching
     // assets, otherwise, you'll only display 1 row in assets table yet there are more than one
     // assets
-    if (assetsEmpty) {
-      this.props.getAssetsAction(activePage, this.state.limit);
-    }
+    this.loadAssets();
 
     this.props.loadAllAssetModels();
     this.props.loadDropdownAssetTypes();
   }
 
+  loadAssets = () => {
+    const { activePage, assetsList, match } = this.props;
+    const { assetStatus } = match.params;
+
+    const assetsEmpty = isEmpty(assetsList);
+    const assetStat = assetStatus === undefined ? '' : assetStatus;
+    const shouldReload = this.props.assetStatus !== assetStat;
+
+    if (shouldReload) {
+      this.props.getAssetsAction(activePage, this.state.limit, null, assetStatus);
+    }
+
+    if (assetsEmpty) {
+      this.props.getAssetsAction(activePage, this.state.limit, null, assetStatus);
+    }
+  };
+
   handleRowChange = (e, data) => {
+    const { activePage, match } = this.props;
+    const { assetStatus } = match.params;
+
     this.setState({ limit: data.value });
     this.props.resetAssets();
-    this.props.getAssetsAction(this.props.activePage, data.value);
+
+    this.props.getAssetsAction(activePage, data.value, null, assetStatus);
   };
 
   handlePaginationChange = (e, { activePage }) => {
+    const { match } = this.props;
+    const { assetStatus } = match.params;
+
     this.props.setActivePage(activePage);
     const currentPageList = this.props.assetsList[`page_${activePage}`];
 
     if (isEmpty(currentPageList)) {
-      this.retrieveAssets(activePage, this.state.limit);
+      this.retrieveAssets(activePage, this.state.limit, assetStatus);
     }
   };
 
-  retrieveAssets = (activePage, limit) => {
+  retrieveAssets = (activePage, limit, assetStatus) => {
     if (checkIfCutoffExceeded(activePage, limit)) {
-      const url = `manage-assets?page=${activePage}&page_size=${limit}`;
+      const url = `manage-assets?page=${activePage}&page_size=${limit}&current_status=${assetStatus}`;

```

Comment:
Move the `constructUrl` in one of my previous comments to `_utils/assets.js` then import the function and use it here